kong/resty/getssl.lua

```diff
@@ -0,0 +1,23 @@
+local ffi = require "ffi"
+local pushssl = require "openssl.ssl".pushffi -- will define SSL* in ffi
+local get_ssl_pointer = require "ngx.ssl".get_ssl_pointer
+
+if get_ssl_pointer == nil then
+  error("Openresty patch missing. See https://github.com/Kong/openresty-patches")
+end
+
+local cast = ffi.cast
+local SSLp = ffi.typeof "SSL*"
+
+local function getssl()
+  local ptr, err = get_ssl_pointer()
+  if not ptr then
+    return nil, err
+  end
+  ptr = cast(SSLp, ptr)
+  return pushssl(ptr)
+end
+
+return {
```

Comment:
style: please respect the 2 lines jump between logical blocks in this codebase - thanks!